checkout/access-checkout-android/src/test/java/com/worldpay/access/checkout/validation/CVVValidatorTest.kt

```diff
@@ -1,289 +1,302 @@
+            CARD_CONFIG_BASIC
+        ))
     }
 
     @Test
     fun `given a pan that doesn't match a brand rule and a cvv with more than valid length against the default rule then should not be valid`() {
-        val cvvValidatorImpl = CVVValidatorImpl(CARD_CONFIG_BASIC)
-
-        assertEquals(Pair(ValidationResult(partial = false, complete = false), null), cvvValidatorImpl.validate("12345", "123456"))
+        assertEquals(Pair(ValidationResult(partial = false, complete = false), null), cvvValidator.validate(
+            "12345",
+            "123456",
+            CARD_CONFIG_BASIC
+        ))
     }
 
     @Test
     fun `given a pan that doesn't match a brand rule and a cvv with length equal to lower bounds of default cvv rule then should be completely valid`() {
-        val cvvValidatorImpl = CVVValidatorImpl(CARD_CONFIG_BASIC)
-
-        assertEquals(Pair(ValidationResult(partial = true, complete = true), null), cvvValidatorImpl.validate("123", "123456"))
+        assertEquals(Pair(ValidationResult(partial = true, complete = true), null), cvvValidator.validate(
+            "123",
+            "123456",
+            CARD_CONFIG_BASIC
+        ))
     }
 
     @Test
     fun `given a pan that doesn't match a brand rule and a cvv with length equal to upper bounds of default cvv rule then should be completely valid`() {
-        val cvvValidatorImpl = CVVValidatorImpl(CARD_CONFIG_BASIC)
-
-        assertEquals(Pair(ValidationResult(partial = false, complete = true), null), cvvValidatorImpl.validate("1234", "123456"))
+        assertEquals(Pair(ValidationResult(partial = false, complete = true), null), cvvValidator.validate(
+            "1234",
+            "123456",
+            CARD_CONFIG_BASIC
+        ))
     }
 
     @Test
     fun `given a pan that doesn't match a brand rule and a cvv with length less than lower bounds of default cvv rule then should be partially valid`() {
-        val cvvValidatorImpl = CVVValidatorImpl(CARD_CONFIG_BASIC)
-
-        assertEquals(Pair(ValidationResult(partial = true, complete = false), null), cvvValidatorImpl.validate("12", "123456"))
+        assertEquals(Pair(ValidationResult(partial = true, complete = false), null), cvvValidator.validate(
+            "12",
+            "123456",
+            CARD_CONFIG_BASIC
+        ))
     }
 
     @Test
     fun `given a pan that doesn't match a brand rule and a cvv with length more than upper bounds of default cvv rule then should be invalid`() {
-        val cvvValidatorImpl = CVVValidatorImpl(CARD_CONFIG_BASIC)
-
-        assertEquals(Pair(ValidationResult(partial = false, complete = false), null), cvvValidatorImpl.validate("123456789", "123456"))
+        assertEquals(Pair(ValidationResult(partial = false, complete = false), null), cvvValidator.validate(
+            "123456789",
+            "123456",
+            CARD_CONFIG_BASIC
+        ))
     }
 
     @Test
     fun `should validate cvv based on default rules when pan is null`() {
```

Comment:
Is there any overlap with other tests?

Reply:
all the tests in this class will be reworked soon, probably is an overlap though. I would not be suprised